service/src/main/java/tds/assessment/repositories/impl/ItemQueryRepositoryImpl.java

```diff
@@ -0,0 +1,133 @@
+package tds.assessment.repositories.impl;
+
+import org.apache.tomcat.jdbc.pool.DataSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.core.namedparam.SqlParameterSource;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import tds.assessment.Item;
+import tds.assessment.ItemConstraint;
+import tds.assessment.ItemProperty;
+import tds.assessment.repositories.ItemQueryRepository;
+
+@Repository
+public class ItemQueryRepositoryImpl implements ItemQueryRepository {
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public ItemQueryRepositoryImpl(final DataSource dataSource) {
+        jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+    }
+
+    public List<Item> findItemsForAssessment(final String assessmentKey) {
```

Comment:
I did briefly consider this, but not sure how much it would really save us in the long run considering the Assessment will likely be pretty heavily cached. I'm fine with either direction.